python/lsst/daf/butler/registry/collections/_base.py

```diff
@@ -271,17 +271,17 @@ def _update(self, manager: CollectionManager, children: CollectionSearch) -> Non
 
     def _load(self, manager: CollectionManager) -> CollectionSearch:
         # Docstring inherited from ChainedCollectionRecord.
-        sql = sqlalchemy.sql.select([
+        sql = sqlalchemy.sql.select(
             self._table.columns.child,
-        ]).select_from(
+        ).select_from(
             self._table
         ).where(
             self._table.columns.parent == self.key
         ).order_by(
             self._table.columns.position
         )
         return CollectionSearch.fromExpression(
-            [manager[row[self._table.columns.child]].name for row in self._db.query(sql)]
+            [manager[row._mapping[self._table.columns.child]].name for row in self._db.query(sql)]
```

Comment:
Very strange to name it with an underscore.  But looking here: https://docs.sqlalchemy.org/en/14/changelog/migration_20.html it says:
> This Row behaves like a named tuple, in that it acts as a sequence but also supports attribute name access, e.g. row.some_column. However, it also provides the previous “mapping” behavior via the special attribute row._mapping, which produces a Python mapping such that keyed access such as row["some_column"] can be used.

So I guess you're supposed to access it with `row.some_column` which seems really inconvenient for dynamically generated column names.

Reply:
Yes, this is a standard issue with `namedtuple`, to avoid clashes with user-defined attributes all public class methods/attributes start with underscore. New SQLAlchemy version is more strict about Row class, it was an odd mix of dict/namedtuple before, not it is just a namedtuple(-like).